spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j2/AmqpAppender.java

```diff
@@ -126,7 +126,7 @@
 	/**
 	 * The template.
 	 */
-	private final RabbitTemplate rabbitTemplate = new RabbitTemplate();
+	private RabbitTemplate rabbitTemplate;
```

Comment:
Has to be `volatile` then?

Reply:
Actually, no, because the `synchronized` block will invalidate the cache.

I re-pushed with the proper `resolves`; another change is needed for the sample when this is merged.

Reply:
Sonar will bark about inconsistent sync, so I added a NOSONAR there.

Reply:
Sorry, I don't see `NOSONAR` in your changes

Reply:
oops (was still local).